command/agent/http.go

```diff
@@ -412,6 +412,8 @@ func (s *HTTPServer) registerHandlers(enableDebug bool) {
 	s.mux.HandleFunc("/v1/plugins", s.wrap(s.CSIPluginsRequest))
 	s.mux.HandleFunc("/v1/plugin/csi/", s.wrap(s.CSIPluginSpecificRequest))
 	s.mux.HandleFunc("/v1/volume/host/", s.wrap(s.HostVolumeSpecificRequest))
+	s.mux.HandleFunc("/v1/volume/claims", s.wrap(s.TaskGroupHostVolumeClaimListRequest))
+	s.mux.HandleFunc("/v1/volume/claim/", s.wrap(s.TaskGroupHostVolumeClaimRequest))
```

Comment:
The existing routes for volumes are a little janky because we lacked the foresight to have a `csi` path component.

The path `/v1/volume/claims` implies a couple of things:
1. It's an endpoint for all the claims on a particular volume, as opposed to `/v1/volumes/claims` which would be the claims on all volumes.
2. It could be doing double-duty for CSI sticky volume claims in the future if we change our mind about that.

I don't feel strongly about (1) in terms of the name, but I also wonder about the utility of "show me all the claims" instead of "show me all the claims, filtered by name, job, etc.". I'm not too worried about (2)... seems like we could manage that without too much trouble so long as we expect them both to have the same shape.

Reply:
Right. So I admit that I yoloed the route here, I just tried using something that felt most intuitive, but I'll agree that `/v1/volumes/claims` is a more accurate route for listing all claims. 

Filtering possibilities would be good, yes. I'll add.

Reply:
[21c8351](https://github.com/hashicorp/nomad/pull/25114/commits/21c8351c169557ffc1b8440b038c743e37098987)